src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java

```diff
@@ -245,44 +246,99 @@ private static String getInstanceName(DetailAST methodCallAst) {
+                        if (isVariableUsed(variableName, methodCallAst)) {
+                            varUsed = true;
+                        }
                     }
                     else {
                         result = false;
                     }
-                }
+                    break;
+                case TokenTypes.VARIABLE_DEF:
+                    final String currentVariableName =
+                            currentSiblingAst.findFirstToken(TokenTypes.IDENT).getText();
+                    isUsedVariableDeclarationFound = variableName.equals(currentVariableName);
+                    break;
 
-            }
-            else if (currentSiblingAst.getType() == TokenTypes.VARIABLE_DEF) {
-                final String currentVariableName =
-                        currentSiblingAst.findFirstToken(TokenTypes.IDENT).getText();
-                isUsedVariableDeclarationFound = variableName.equals(currentVariableName);
-            }
-            else {
-                result = currentSiblingAst.getType() == TokenTypes.SEMI;
+                default:
+                    result = currentSiblingAst.getType() == TokenTypes.SEMI;
+                    break;
             }
             currentSiblingAst = currentSiblingAst.getPreviousSibling();
         }
+        if (variableUsageAst != null) {
+            result = result && varUsed;
+        }
         return result;
     }
 
+    /**
+     * Checks if the provided instance names are equal.
+     *
+     * @param initInstanceName The init instance name.
+     * @param instanceName The instance name.
+     * @return true if the instance names is equal to initInstanceName.
+     */
+    private static boolean areInstanceNamesEqual(String initInstanceName, String instanceName) {
+        return !instanceName.isEmpty()
+                        && instanceName.equals(initInstanceName);
+    }
+
+    /**
+     * Checks if the specified variable name matches any identifier.
+     *
+     * @param variableName The name of the variable to search for within the AST.
+     * @param ast The root of the (AST) to search within.
+     * @return true if the variable name matches any identifier within the AST;
+     */
+    private static boolean isVariableUsed(String variableName, DetailAST ast) {
+        final List<DetailAST> identAst = getIdentAst(ast);
+        return identAst.stream()
+            .map(DetailAST::getText)
+            .anyMatch(name -> name.equals(variableName));
+    }
+
+    /**
+     * Return the list of ident AST.
+     *
+     * @param ast The root of the (AST) to search within.
+     * @return list of ident ast within the given AST;
+     */
+    private static List<DetailAST> getIdentAst(DetailAST ast) {
+        DetailAST curNode = ast;
+        final List<DetailAST> identList = new ArrayList<>();
+        while (curNode.getType() != TokenTypes.SEMI) {
+            if (curNode.getType() == TokenTypes.IDENT) {
```

Comment:
A quick update on progression of this pr currently the ci of pitest is red because this is mutated 
```
Source File: "VariableDeclarationUsageDistanceCheck.java"
Class: "com.puppycrawl.tools.checkstyle.checks.coding.VariableDeclarationUsageDistanceCheck"
Method: "getIdentAst"
Line Contents: "if (curNode.getType() == TokenTypes.IDENT) {"
Mutator: "org.pitest.mutationtest.engine.gregor.mutators.RemoveConditionalMutator_EQUAL_IF"
Description: "removed conditional - replaced equality check with true"
Line Number: 329
```

I do have the test case to kill this the code like 
```
        void method() {
            Test non = new Test();
            String test = "check"; // violation ''
            non.test("b");
            non.test("b");
            non.test("a");
            non.test("a");
            non.test("a");
            non.test("a");
            non.test("a");
            System.out.println(test);
        }

```
Can kill this mutation

Reply:
but currently, we have again some other bug probably in calculating distance.
so that the code below is not showing the violation
```
kevin@kevin-Inspiron-15-5510:~/Desktop/check_style$ javac Test.java 
kevin@kevin-Inspiron-15-5510:~/Desktop/check_style$ 
kevin@kevin-Inspiron-15-5510:~/Desktop/check_style$ cat Test.java 
public class Test {

  public Test test(Object a) {
    return this;
  }

  void method() {
    Test non = new Test();
    String name = "Checkstyle";
    String test = "check"; // violation expected
    non.test("b");
    if (10 > 20) {
      System.out.println("maths rule changed");
    }
    if (20 > 10) {
      System.out.println("good");
    }
    System.out.println(name);
    System.out.println(test);
  }
  class subclass extends Test {
  }
}
kevin@kevin-Inspiron-15-5510:~/Desktop/check_style$ cat config.xml 
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.2//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_2.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="VariableDeclarationUsageDistance">
            <property name="allowedDistance" value="1"/>
        </module>
    </module>
</module>
kevin@kevin-Inspiron-15-5510:~/Desktop/check_style$ java -jar /home/kevin/Downloads/checkstyle-10.14.0-all.jar  -c config.xml Test.java 
Starting audit...
[ERROR] /home/kevin/Desktop/check_style/Test.java:9:5: Distance between variable 'name' declaration and its first usage is 4, but allowed 1.  Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value). [VariableDeclarationUsageDistance]
Audit done.
Checkstyle ends with 1 errors.
kevin@kevin-Inspiron-15-5510:~/Desktop/check_style$ 
```
I expect a violation on line no 10.

I will figure out this ASAP and update this pr